app/src/main/java/ch/epfl/sweng/runpharaa/FragmentFollowing.java

```diff
@@ -59,6 +59,7 @@ public void onItemClick(CardItem item) {
 
                 List<User> users = UserDatabaseManagement.initFollowingFragment(data);
                 for (User u : users) {
+                    if(u == null) continue; //TODO: this check should not be necessary, please check why deserialized users are null
```

Comment:
Just pulled your branch, working on it.

Reply:
The users obtained from deserialized should never be null. The only way to have a null User in the List<User> is when the deserialized() method from User failed. 
In this case, an error will appear with the tag "Deserialization Error" in the logcat with the error message.

I think that the check for null here is not a bad thing to avoid crashing the application if that happens :)

Reply:
I think there is a problem with this piece of code first (line 94 of User.class):
We first catch any Exception (which is not good practice), and we return null without stopping the execution (which just move the problem later)...

```java
catch (Exception e) {
            Log.d("Deserialization Error", e.toString());
            return null;
        }
```